src/utils.ts

```diff
@@ -259,22 +261,56 @@ export const parseCborData = (buf: Uint8Array): IPNSRecordData => {
 }
 
 /**
- * Normalizes the given record value. It ensures it is a string starting with '/'.
- * If the given value is a cid, the returned path will be '/ipfs/{cid}'.
+ * Normalizes the given record value. It ensures it is a PeerID, a CID or a
+ * string starting with '/'. PeerIDs become `/ipns/${peerId}`, CIDs become
+ * `/ipfs/${cidAsV1}`.
  */
-export const normalizeValue = (value: string | Uint8Array): string => {
-  const str = typeof value === 'string' ? value : uint8ArrayToString(value)
+export const normalizeValue = (value?: CID | PeerId | string | Uint8Array): string => {
```

Comment:
nit: we want to remove use of PeerIDs in IPNS context, as it causes bugs where someone tries to parse `12..` one as CID ([example](https://cid.ipfs.tech/#12D3KooWHj48PbuAPUf9LeovBJZEXn2A4rseW5z9wbTBwvM82nAq))  and we've been normalizing to CIDv1 with libp2p-key and base36 (this way it looks the same on subdomains).

@achingbrain would it be ok to switch `/ipns/` paths to use CIDv1 with libp2p-key and base36?

Reply:
@achingbrain I think the goal moving forward into the future is to always refer to `/ipns/{libp2p-encoded-cid}` and not as a base56btc: https://specs.ipfs.tech/ipns/ipns-record/#string-representation, including on the gateway.

Reply:
Ok, as long as we're fine with it being `/ipns/{maybe-CID,maybe-base58btcpeerid}` I guess?